pkg/framework/plugins/defaultevictor/defaultevictor.go

```diff
@@ -77,139 +73,161 @@ func New(args runtime.Object, handle frameworktypes.Handle) (frameworktypes.Plug
+	effectiveProtectedPolicies := getEffectiveProtectedPolicies(args)
+
+	if err := applyEffectiveProtectedPolicies(ev, effectiveProtectedPolicies, handle); err != nil {
+		return fmt.Errorf("failed to apply effective protected policies: %w", err)
 	}
-	if !defaultEvictorArgs.EvictLocalStoragePods {
-		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
-			if utils.IsPodWithLocalStorage(pod) {
-				return fmt.Errorf("pod has local storage and descheduler is not configured with evictLocalStoragePods")
-			}
-			return nil
-		})
+
+	// Step 2: Add label selector constraint
+	if err := addLabelSelectorConstraint(ev, args.LabelSelector); err != nil {
+		return fmt.Errorf("failed to add label selector constraint: %w", err)
 	}
-	if !defaultEvictorArgs.EvictDaemonSetPods {
-		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
-			ownerRefList := podutil.OwnerRef(pod)
-			if utils.IsDaemonsetPod(ownerRefList) {
-				return fmt.Errorf("pod is related to daemonset and descheduler is not configured with evictDaemonSetPods")
-			}
-			return nil
-		})
+
+	// Step 3: Add min replicas constraint
+	if err := addMinReplicasConstraint(ev, args.MinReplicas, handle); err != nil {
+		return fmt.Errorf("failed to add min replicas constraint: %w", err)
 	}
-	if defaultEvictorArgs.IgnorePvcPods {
-		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
-			if utils.IsPodWithPVC(pod) {
-				return fmt.Errorf("pod has a PVC and descheduler is configured to ignore PVC pods")
-			}
-			return nil
-		})
+
+	// Step 4: Add min pod age constraint
+	if err := addMinPodAgeConstraint(ev, args.MinPodAge); err != nil {
+		return fmt.Errorf("failed to add min pod age constraint: %w", err)
 	}
-	selector, err := metav1.LabelSelectorAsSelector(defaultEvictorArgs.LabelSelector)
-	if err != nil {
-		return nil, fmt.Errorf("could not get selector from label selector")
+	return nil
+}
+
+var AllPolicies = []PodProtectionPolicy{
+	PodsWithLocalStorage,
+	DaemonSetPods,
+	SystemCriticalPods,
+	FailedBarePods,
+	PodsWithPVC,
+	PodsWithoutPDB,
+}
+
+// applyEffectiveProtectedPolicies applies all relevant protection policies to the plugin.
+// For each policy:
+// - If it's in the list, we disallow eviction (shouldEvict = false).
+// - If it's NOT in the list, we allow eviction (shouldEvict = true).
+func applyEffectiveProtectedPolicies(ev *DefaultEvictor, policies []PodProtectionPolicy, handle frameworktypes.Handle) error {
+	policyMap := make(map[PodProtectionPolicy]bool)
+	for _, policy := range policies {
+		policyMap[policy] = true
 	}
-	if defaultEvictorArgs.LabelSelector != nil && !selector.Empty() {
-		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
-			if !selector.Matches(labels.Set(pod.Labels)) {
-				return fmt.Errorf("pod labels do not match the labelSelector filter in the policy parameter")
-			}
-			return nil
-		})
+
+	for _, policy := range AllPolicies {
```

Comment:
The same question here. Can you more elaborate on why iterating through `AllPolicies` is better than directly invoking the individual eviction constraint functions? I.e.
```go
addEvictionConstraintsForLocalStoragePods(ev, !policyMap[PodsWithLocalStorage])
addEvictionConstraintsForDaemonSetPods(ev, !policyMap[DaemonSetPods])
...

```